zzk/service/hostregistry.go

```diff
@@ -250,8 +251,21 @@ func (h *HostRegistryListener) Spawn(cancel <-chan interface{}, hostid string) {
 						// soon as possible.
 						select {
 						case <-h.isOnline:
-							count := DeleteHostStates(h.conn, h.poolid, hostid)
+							// Only reschedule services without address
+							// assignments.
+							count := DeleteHostStatesWhen(h.conn, h.poolid, hostid, func(s *State) bool {
+								return s.DesiredState == service.SVCStop || !s.Static
+							})
 							logger.WithField("unscheduled", count).Warn("Host is experiencing an outage.  Cleaned up orphaned nodes")
+
+							// To prevent a tight loop, wait for something to
+							// happen.
+							select {
```

Comment:
Is there no timeout here needed?  Why not go back to the top of this inner select?